src/app/clusters/commodity-price-server/commodity-price-server.cpp

```diff
@@ -246,6 +246,15 @@ Instance::GetDetailedForecastRequest(BitMask<CommodityPriceDetailBitmap> details
     size_t count      = 0;
     size_t bufferSize = mPriceForecast.size();
 
+    if (bufferSize == 0)
+    {
+        /* Special case when no forcast entries exist - calling calloc(0) returns NULL
+           and results in an error on some platforms */
+        forecastList = DataModel::List<const Structs::CommodityPriceStruct::Type>(Span<Structs::CommodityPriceStruct::Type>());
```

Comment:
Why do you need the Span there?  Just:
```suggestion
        forecastList = DataModel::List<const Structs::CommodityPriceStruct::Type>();
```

Reply:
Ok - change applied